### PR TITLE
PMM-6014 adding parameters for the builds

### PR DESCRIPTION
**Pull request by denisok**
Adding more parameters for the PMM server build to be able manually
release from the release branch.

DO NOT MERGE

### CODE REVIEW

**Comment by denisok** [comment]
need to figure out what to do with images (ovf, ami and etc). And test these changes especially those that modifies env.
Adding review to see if anything dirty in place

**Comment by denisok** [comment]
yeah, we need to customize this as well: https://github.com/Percona-Lab/percona-images/blob/master/packer/pmm2.json#L76
That we could customize with dev branch as manual step and make changes in the branch in a worst case.

File: pmm/pmm2-server-autobuild.groovy [hunk 1]
```diff
@@ -52,8 +57,17 @@ pipeline {
                     popd
 
                     git rev-parse --short HEAD > shortCommit
-                    echo "UPLOAD/${DESTINATION}/${JOB_NAME}/pmm/\$(cat VERSION)/${GIT_BRANCH}/\$(cat shortCommit)/${BUILD_NUMBER}" > uploadPath
+                    echo "UPLOAD/pmm2-components/yum/${DESTINATION}/${JOB_NAME}/pmm/\$(cat VERSION)/${GIT_BRANCH}/\$(cat shortCommit)/${BUILD_NUMBER}" > uploadPath
+                    cat VERSION > versionTag
                 '''
+
+                script {
+                    def versionTag = readFile('versionTag').trim()
```
Comment (BupycHuk) [inline pmm/pmm2-server-autobuild.groovy:65]: maybe we can just read `VERSION`?

**Review by BupycHuk — APPROVED** [review]

**Comment by BupycHuk** [comment]
> yeah, we need to customize this as well: https://github.com/Percona-Lab/percona-images/blob/master/packer/pmm2.json#L76
That we could customize with dev branch as manual step and make changes in the branch in a worst case.
> 
> And we need to customize image names so RC could be visible with version and rc number on the list of other images.

What code calls this ansible script? as far as I see ansible lets us to pass custom parameters.

**Comment by denisok** [comment]
> > yeah, we need to customize this as well: https://github.com/Percona-Lab/percona-images/blob/master/packer/pmm2.json#L76
> > That we could customize with dev branch as manual step and make changes in the branch in a worst case.
> > And we need to customize image names so RC could be visible with version and rc number on the list of other images.
> 
> What code calls this ansible script? as far as I see ansible lets us to pass custom parameters.

packer calls it to provision images. Packer is started by build jobs like:
https://pmm.cd.percona.com/job/pmm2-ami/

File: pmm/pmm2-server-autobuild.groovy (hunk 1 above)
Comment (denisok) [reply pmm/pmm2-server-autobuild.groovy:65]: should work but it doesn't. Neither mine, nor just reading from VERSION

**Comment by denisok** [comment]
@EvgeniyPatlan I don't understand why it doesn't work. But maybe it is too late. I have created PMM-6014_parameterizing-build-test and https://pmm.cd.percona.com/job/pmm2-server-autobuild-test/ test job.

My problem is  `def versionTag = readFile('versionTag').trim()` it doesn't look to work as `echo "latest tag: ${versionTag} build: ${BUILD_NUMBER} docker tag: ${DOCKER_LATEST_TAG}"` returns:  `latest tag:  build: 9 docker tag: dev-latest`
same if I just read `VERSION` file instead of `versionTag`.

maybe you have another idea, or could see what the problem is.

**Comment by puneet0191** [comment]
@denisok I realized the branch specified in Jenkins job and the branch being used for this PR are different, I was not able to see my changes also, I have updated it and now trying a different method.

**Comment by denisok** [comment]
> @denisok I realized the branch specified in Jenkins job and the branch being used for this PR are different, I was not able to see my changes also, I have updated it and now trying a different method.

yeah, and please comment a build steps ) they take too much time and it was too late for me to realize that :)

File: pmm/pmm2-server-autobuild.groovy [hunk 2]
```diff
@@ -50,10 +55,21 @@ pipeline {
                     git lfs pull
                     git lfs checkout
                     popd
+                    ls -la
```
Comment (atymchuk) [inline pmm/pmm2-server-autobuild.groovy:58]: Add `rm git-lfs-linux-amd64-v2.7.1.tar.gz` on line 45

File: pmm/pmm2-server-autobuild.groovy [hunk 3]
```diff
@@ -50,10 +55,21 @@ pipeline {
...
+                    pwd
 
                     git rev-parse --short HEAD > shortCommit
-                    echo "UPLOAD/${DESTINATION}/${JOB_NAME}/pmm/\$(cat VERSION)/${GIT_BRANCH}/\$(cat shortCommit)/${BUILD_NUMBER}" > uploadPath
+                    echo "UPLOAD/pmm2-components/yum/${DESTINATION}/${JOB_NAME}/pmm/\$(cat VERSION)/${GIT_BRANCH}/\$(cat shortCommit)/${BUILD_NUMBER}" > uploadPath
+                    cat VERSION > versionTag
```
Comment (atymchuk) [inline pmm/pmm2-server-autobuild.groovy:63]: This is redundant

File: pmm/pmm2-server-autobuild.groovy (hunk 2 above)
Comment (puneet0191) [reply pmm/pmm2-server-autobuild.groovy]: Fixed, thanks

File: pmm/pmm2-server-autobuild.groovy (hunk 3 above)
Comment (puneet0191) [reply pmm/pmm2-server-autobuild.groovy]: Yeap, removed it. thanks

**Review by puneet0191 — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**